class.jetpack.php

```diff
@@ -568,28 +576,36 @@ static function delete_active_modules() {
 	public function add_configure_hook() {
 		global $wp_filter;
 
-		$current_priority = has_filter( 'plugins_loaded', array( $this, 'configure' ) );
-		if ( false !== $current_priority ) {
-			remove_action( 'plugins_loaded', array( $this, 'configure' ), $current_priority );
+		// If there is no handler set for the plugins_loaded hook yet, set it and return.
+		if ( ! isset( $wp_filter['plugins_loaded'] ) ) {
+			add_action( 'plugins_loaded', array( $this, 'configure' ), 0 );
+			return;
 		}
 
-		$taken_priorities = array_map( 'intval', array_keys( $wp_filter['plugins_loaded']->callbacks ) );
-		sort( $taken_priorities );
+		$highest_priority = min( array_keys( $wp_filter['plugins_loaded']->callbacks ) );
 
-		$first_priority = array_shift( $taken_priorities );
+		// If our hook is set, and it's still at highest priority, we don't need to do anything.
+		if (
+			isset( $this->highest_priority )
 
-		if ( defined( 'PHP_INT_MAX' ) && $first_priority <= - PHP_INT_MAX ) {
-			trigger_error( // phpcs:ignore
-				/* translators: plugins_loaded is a filter name in WordPress, no need to translate. */
-				__( 'A plugin on your site is using the plugins_loaded filter with a priority that is too high. Jetpack does not support this, you may experience problems.', 'jetpack' ), // phpcs:ignore
-				E_USER_NOTICE
-			);
-			$new_priority = - PHP_INT_MAX;
-		} else {
-			$new_priority = $first_priority - 1;
+			// Priority is in reverse order, so less is higher.
+			&& $this->highest_priority <= $highest_priority
+		) {
+			return;
 		}
 
-		add_action( 'plugins_loaded', array( $this, 'configure' ), $new_priority );
+		$actions_on_highest_priority = $wp_filter['plugins_loaded']->callbacks[ $highest_priority ];
+
+		foreach ( $actions_on_highest_priority as $action ) {
+			remove_action( 'plugins_loaded', $action['function'], $highest_priority );
+		}
+
+		$this->highest_priority = $highest_priority;
+		add_action( 'plugins_loaded', array( $this, 'configure' ), $highest_priority );
+
+		foreach ( $actions_on_highest_priority as $action ) {
+			add_action( 'plugins_loaded', $action['function'], $highest_priority, $action['accepted_args'] );
+		}
```

Comment:
If `configure` is already hooked at some priority and needs to be moved to a higher priority, we need to remove the original hook, right? I don’t think that’s being done.

Reply:
Yes, you're right, great catch! I have fixed that and added a test against that.